src/main/scala/lms/core/stub.scala

```diff
@@ -806,6 +806,11 @@ trait LiftPrimitives {
 object PrimitiveTypeLess {
   import BaseTypeLess._
 
+  def CAST_HELPER(x: TOP, mX: Manifest[_], mY: Manifest[_], pos: SourceContext)(implicit __pos: SourceContext): TOP = {
```

Comment:
- if `x.t` has to be the same manifest as `mX`, why `mX` has to be provided? 
- why there is an explicit `pos` and an implicit `pos`?
- also wondering if this can be implemented in a more idiomatic way, eg, in the `class TOP` we define a method `castTo`:
```
def castTo(mY: Manifest[_])(implicit pos: SourceContext) = { ... }
```
or
```
def castTo[Y: Manifest](implicit pos: SourceContext) = { val mY = manifest[T]; ... }
```